src/uwtools/config/formats/base.py

```diff
@@ -35,6 +36,11 @@ def __init__(self, config: Union[dict, OptionalPath] = None) -> None:
         else:
             self._config_file = str(config) if config else None
             self.update(self._load(self._config_file))
+        if self.get_depth_threshold() and self.depth != self.get_depth_threshold():
```

Comment:
Good idea to just init with a check too since we're already looking everywhere else

Reply:
Next steps would be to intercept write access to the `data` attribute and perform this check there, too. If we were able to implement that correctly, I think we'd only need checks 1. at init time and 2. at write time. I think that would ensure correct depth at all times, and we could remove other checks, as they'd be redundant.